Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -109,6 +109,34 @@
 
 using namespace amrex;
 
+template <int... ctr>
+struct CompiletimeOptions {};
+
+template <int ctr>
+struct CTOption {
+    static constexpr int value = ctr;
+};
+
+template <class F, int... Op>
+void ParallelFor(Long N, F&& f, CompiletimeOptions<Op...>, int runtime_option)
+{
+    int option_miss = 0;
+    (
+        (
+            Op == runtime_option ?
+              (
+                  amrex::ParallelFor(N, [f] AMREX_GPU_DEVICE (Long i) {
+                      f(i, CTOption<Op>{});
+                  })
+              )
+            : (
+                ++option_miss, void()
+              )
+        ),...
+    );
+    AMREX_ASSERT(option_miss < sizeof...(option_pack));
```

Comment:
```suggestion
    AMREX_ASSERT(option_miss + 1 == sizeof...(Op));
```
Should be a bit safer this way